modules/dop/endpoints/workbench.go

```diff
@@ -77,102 +66,14 @@ func (e *Endpoints) GetWorkbenchData(ctx context.Context, r *http.Request, vars
 	}
 	proTotal, unDonePros := resp.Total, resp.List
 	res.Data.TotalProject = proTotal
-
-	nowTime := time.Date(time.Now().Year(), time.Now().Month(), time.Now().Day(), 0, 0, 0, 0, time.Now().Location())
-	tomorrow := nowTime.Add(time.Hour * time.Duration(24))
-	twoDay := nowTime.Add(time.Hour * time.Duration(24*2))
-	sevenDay := nowTime.Add(time.Hour * time.Duration(24*7))
-	thirtyDay := nowTime.Add(time.Hour * time.Duration(24*30))
-	timeList := [][]int64{
-		{0, 0}, // not specified
-		{1, nowTime.Add(time.Second * time.Duration(-1)).Unix()},                 // expired
-		{nowTime.Unix(), tomorrow.Add(time.Second * time.Duration(-1)).Unix()},   // today expired
-		{tomorrow.Unix(), twoDay.Add(time.Second * time.Duration(-1)).Unix()},    // tomorrow expired
-		{twoDay.Unix(), sevenDay.Add(time.Second * time.Duration(-1)).Unix()},    // seven day expired
-		{sevenDay.Unix(), thirtyDay.Add(time.Second * time.Duration(-1)).Unix()}, // thirty day expired
-		{thirtyDay.Unix(), 0}, //feature expired
-	}
-	expireDays := []string{"unspecified", "expired", "oneDay", "tomorrow", "sevenDay", "thirtyDay", "feature"}
-
 	res.Data.List = make([]apistructs.WorkbenchProjectItem, 0)
+
 	for _, v := range unDonePros {
 		var issueItem apistructs.WorkbenchProjectItem
-		issueItem.IssueList = make([]apistructs.Issue, 0)
-		issueReq := apistructs.IssuePagingRequest{
-			OrgID:    int64(workReq.OrgID),
-			PageNo:   1,
-			PageSize: uint64(workReq.IssueSize),
-			IssueListRequest: apistructs.IssueListRequest{
-				ProjectID:    uint64(v.ID),
-				StateBelongs: stateBelongs,
-				Assignees:    []string{userID.String()},
-				External:     true,
-				OrderBy:      "plan_finished_at asc, FIELD(priority, 'URGENT', 'HIGH', 'NORMAL', 'LOW')",
-				Priority: []apistructs.IssuePriority{
-					apistructs.IssuePriorityUrgent,
-					apistructs.IssuePriorityHigh,
-					apistructs.IssuePriorityNormal,
-					apistructs.IssuePriorityLow,
-				},
-				Type: []apistructs.IssueType{
-					apistructs.IssueTypeRequirement,
-					apistructs.IssueTypeBug,
-					apistructs.IssueTypeTask,
-				},
-				Asc: true,
-			},
-		}
-		issues, total, err := e.issue.Paging(issueReq)
+		issueItem, err := e.workBench.GetUndoneProjectItem(userID.String(), workReq.IssueSize, v)
```

Comment:
let me describe by pseudo-code, what you do now:

```
for v := range projects {
  for t := range type of issue state { // 7 types
      // this for is concurrently, but only calculate one project
  }
}
```
at the result, if there are 10 projects, it will have 10 x 7 = 70 times sql, as it run as concurrently, in user experience, it will have 10 times slow.

And what you can improve, I suggest you should:
```
for t := range type of issue state {
   bulk query and sum all projects (10 projects), only for this type
}
```
and this `for` can be concurrently, so as the result, there will be 7 sql no matter how many projects. In user experience, always fast like one sql.

Reply:
can`t query issue num by project list, becase query issue num should stateID list first, different project`s state list are different